### ISSUE TITLE
Do not specify an exact version of peerDependencies

### ISSUE THREAD

**Issue by rkeene** [mbrn/material-table#2817]
**Bug Description**
As of #2510 (included in release 1.69.2, apparently), peer dependencies are specified as an exact version -- that is, to use "material-table" you must have *EXACTLY* React 16.8.4.   In previous releases, you only needed a version that was "compatible and at least" ^16.8.4.

**To Reproduce**

1. Install React 16.8.5
2. Attempt to install material-table 1.69.2
3. Notice failure

**Expected behavior**
material-table 1.69.2 should be able to install with versions of React other than exactly 16.8.4.

**Screenshots**
```
$ npm install --save-exact react@16.8.5 react-dom@16.8.5

added 8 packages, and audited 9 packages in 3s

found 0 vulnerabilities
$ npm install --save-exact material-table@1.69.2
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: demo@0.1.0
npm ERR! Found: react@16.8.5
npm ERR! node_modules/react
npm ERR!   react@"16.8.5" from the root project
npm ERR!   peer react@"^16.8.0" from @material-ui/core@4.0.1
npm ERR!   node_modules/@material-ui/core
npm ERR!     peer @material-ui/core@"4.0.1" from material-table@1.69.2
npm ERR!     node_modules/material-table
npm ERR!       material-table@"1.69.2" from the root project
npm ERR!   1 more (react-dom)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"16.8.4" from material-table@1.69.2
npm ERR! node_modules/material-table
npm ERR!   material-table@"1.69.2" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

**Comment by rkeene** [comment]
Working with material-table 1.69.1, which does not specify an exact version of React as a peer dependency:

```
$ npm install react@16.8.5 react-dom@16.8.5

added 8 packages, and audited 9 packages in 1s

found 0 vulnerabilities
$ npm install material-table@1.69.1

added 73 packages, and audited 82 packages in 8s

7 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
```

**Comment by rkeene** [comment]
Fixed by #2797
